src/gpt_j/gpt_j_model.rs

```diff
@@ -0,0 +1,808 @@
+    ///   - `all_hidden_states` - `Option<Vec<Tensor>>` of length *num_hidden_layers* with shape (*batch size*, *sequence_length*, *hidden_size*)
+    ///   - `all_attentions` - `Option<Vec<Tensor>>` of length *num_hidden_layers* with shape (*batch size*, *sequence_length*, *hidden_size*)
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use tch::{nn, Device, Tensor, no_grad};
+    /// # use rust_bert::Config;
+    /// # use std::path::Path;
+    /// # use tch::kind::Kind::{Int64, Double};
+    /// use rust_bert::gpt_j::{GptJConfig, GptJModel};
+    /// # let config_path = Path::new("path/to/config.json");
+    /// # let vocab_path = Path::new("path/to/vocab.txt");
+    /// # let device = Device::Cpu;
+    /// # let vs = nn::VarStore::new(device);
+    /// # let config = GptJConfig::from_file(config_path);
+    /// # let gpt_j_model: GptJModel = GptJModel::new(&vs.root(), &config);
+    /// let (batch_size, sequence_length, past_sequence_length) = (64, 128, 56);
+    /// let input_tensor = Tensor::rand(&[batch_size, sequence_length], (Int64, device));
+    /// let mut past: Vec<Tensor> = Vec::with_capacity(config.n_layer as usize);
+    /// for _ in 0..config.n_layer as usize {
+    ///     past.push(Tensor::rand(
+    ///         &[
+    ///             2,
+    ///             batch_size,
+    ///             config.n_head,
+    ///             past_sequence_length,
+    ///             config.n_embd / config.n_head,
+    ///         ],
+    ///         (Double, device),
+    ///     ))
+    /// }
+    /// let attention_mask = Tensor::zeros(&[batch_size, sequence_length], (Int64, device));
+    /// let token_type_ids = Tensor::ones(&[batch_size, sequence_length], (Int64, device));
+    ///
+    /// let model_output = no_grad(|| {
+    ///     gpt_j_model
+    ///         .forward_t(
+    ///             Some(&input_tensor),
+    ///             Some(&past),
+    ///             Some(&attention_mask),
+    ///             Some(&token_type_ids),
+    ///             None,
+    ///             None,
+    ///             false,
+    ///         )
+    ///         .unwrap()
+    /// });
+    /// ```
+    pub fn forward_t(
+        &self,
+        input_ids: Option<&Tensor>,
+        layer_past: Option<Vec<Option<LayerState>>>,
+        attention_mask: Option<&Tensor>,
+        token_type_ids: Option<&Tensor>,
+        _position_ids: Option<&Tensor>,
+        input_embeds: Option<&Tensor>,
+        train: bool,
+    ) -> Result<GptJModelOutput, RustBertError> {
+        let (calc_input_embeddings, _input_size, _device) =
+            process_ids_embeddings_pair(input_ids, input_embeds, &self.wte)?;
+
+        let input_embeddings =
+            input_embeds.unwrap_or_else(|| calc_input_embeddings.as_ref().unwrap());
+
+        let (layer_past, _layer_past_length) = match layer_past {
+            Some(value) => {
+                assert_eq!(
```

Comment:
Since this method returns a `Result` you could raise an `Error` rather than panicking

Reply:
Done